go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -234,17 +241,14 @@ func tryMergeSubQuery(ctx planningContext, outer, subq queryTree, subQueryInner
 	}
 }
 
-func rewriteSubqueryDependenciesForJoin(a *routeTree, outerTree *joinTree, subQueryInner *abstract.SubQueryInner, ctx planningContext) error {
+// outerTree is the joinTree within whose children the subquery lives in
+// the child of joinTree which does not contain the subquery is the otherTree
+func rewriteSubqueryDependenciesForJoin(otherTree queryTree, outerTree *joinTree, subQueryInner *abstract.SubQueryInner, ctx planningContext) error {
 	// first we find the other side of the tree by comparing the tableIDs
 	// other side is RHS if the subquery is in the LHS, otherwise it is LHS
-	otherTree := outerTree.lhs
-	if a.tableID() == outerTree.lhs.tableID() {
-		otherTree = outerTree.rhs
-	}
-
 	var rewriteError error
 	// go over the entire where expression in the subquery
-	sqlparser.Rewrite(subQueryInner.SelectStatement.Where.Expr, func(cursor *sqlparser.Cursor) bool {
+	sqlparser.Rewrite(subQueryInner.SelectStatement, func(cursor *sqlparser.Cursor) bool {
```

Comment:
Why not starting the rewrite at `subQueryInner.SelectStatement.Where.Expr`? We could do a `nil` check just before the rewrite to avoid panic when the subquery's inner does not have a where clause.

Reply:
It's not just about predicates though. We changes this because we were not rewriting projected expressions on the subquery. We need to rewrite all expressions in the subquery